samza-core/src/main/java/org/apache/samza/job/metadata/JobMetadataChange.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.job.metadata;
+
+/**
+ * Provides granularity into changes in job metadata (e.g. new job model, new config).
+ */
+public enum JobMetadataChange {
```

Comment:
Can we move this under samza-api along with `JobCoordinatorMetadata`?

Reply:
To me, in the current state of the code, this doesn't seem like an actual API at this point. Implementations of coordinators use this object, but if you use that categorization for "API", then you also need to include any of the other classes that could be used by a coordinator (e.g. `StreamMetadataCache`, `StreamPartitionCountMonitor`, `JobCoordinatorMetadataManager`, etc.). Therefore, just moving a couple classes doesn't really do much.
I actually do think there is a lot of opportunity for refactoring the classes into modules such that these components do go into a more explicit "API" module, but that is a much bigger scope than this PR.

Reply:
I agree about the lack of internal api module and treating `samza-api` as an interim path may not be ideal but the only option. 

I was mostly looking at this from the point of consistency given this `model` is sort of related to `JobCoordinatorMetadata` and that resides in `samza-api`.  Even if we were to introduce an internal api module down the lane, its easy to discover the related classes under the same module and reduce the surface of change to one self contained module for these classes alone if not for all the model classes in samza.

Reply:
Ok, makes sense, I'll move it.